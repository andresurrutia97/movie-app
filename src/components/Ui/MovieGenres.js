import React from 'react';
import PropTypes from 'prop-types';

const MovieGenres = ({ genres }) => {
  const movieGenres = genres.map((genre) => genre.name);
  return <div className="my-4">{movieGenres.join(', ')}</div>;
};

MovieGenres.propTypes = {
  genres: PropTypes.array,
};

MovieGenres.defaultProps = {
  genres: [],
};

export default MovieGenres;
