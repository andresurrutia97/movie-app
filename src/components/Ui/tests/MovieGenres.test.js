import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import MovieGenres from '../MovieGenres';

describe('<MovieGenres/>', () => {
  const genres = [{ name: 'horror' }, { name: 'comedy' }];
  const component = <MovieGenres genres={genres} />;

  it('should render correctly ', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain an element with a string that has the name of the genres ', () => {
    const wrapper = shallow(component);
    expect(wrapper.props().children).toMatch(new RegExp('horror'));
    expect(wrapper.props().children).toMatch(new RegExp('comedy'));
  });
});
