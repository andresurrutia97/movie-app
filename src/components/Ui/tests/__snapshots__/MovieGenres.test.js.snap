// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieGenres/> should render correctly  1`] = `
<div
  className="my-4"
>
  horror, comedy
</div>
`;
